drivers/rf-transceiver/hmc630x/iio_hmc630x.c

```diff
@@ -0,0 +1,378 @@
+		enum hmc6301_lna_gain lna_gain;
+		ret = hmc6301_get_lna_gain(d, &lna_gain);
+		if (ret)
+			return ret;
+
+		val = lna_gain;
+		break;
+	case HMC630X_IIO_ATTR_BB_ATTN1:
+	case HMC630X_IIO_ATTR_BB_ATTN2:
+		enum hmc6301_bb_attn attn1, attn2;
+		ret = hmc6301_get_bb_attn(d, &attn1, &attn2);
+		if (ret)
+			return ret;
+
+		if (priv == HMC630X_IIO_ATTR_BB_ATTN1)
+			val = attn1;
+		else
+			val = attn2;
+		break;
+	default:
+		return -EINVAL;
+	};
+
+	return iio_format_value(buf, len, IIO_VAL_INT, 1, &val);
+}
+
+static int hmc630x_iio_write_attr(void *device, char *buf,
+				  uint32_t len, const struct iio_ch_info *channel,
+				  intptr_t priv)
+{
+	int ret = 0;
+	int32_t val;
+	struct hmc630x_iio_dev *iiodev = (struct hmc630x_iio_dev *)device;
+	struct hmc630x_dev *d = (struct hmc630x_dev *)iiodev->dev;
+
+	ret = iio_parse_value(buf, IIO_VAL_INT, &val, NULL);
+	if (ret < 0)
+		return ret;
+
+	switch (priv) {
+	case HMC630X_IIO_ATTR_ENABLED:
+		ret = hmc630x_set_enable(d, (bool)val);
+		break;
+	case HMC630X_IIO_ATTR_VCO:
+		ret = hmc630x_set_vco(d, (uint64_t)val * 1000);
+		break;
+	case HMC630X_IIO_ATTR_IF_ATTN:
+		ret = hmc630x_set_if_attn(d, (uint8_t)val);
+		break;
+	case HMC630X_IIO_ATTR_TEMP_EN:
+		ret = hmc630x_set_temp_en(d, (bool)val);
+		break;
+	case HMC630X_IIO_ATTR_RF_ATTN:
+		ret = hmc6300_set_rf_attn(d, (uint8_t)val);
+		break;
+	case HMC630X_IIO_ATTR_RF_LNA_GAIN:
+		ret = hmc6301_set_lna_gain(d, (enum hmc6301_lna_gain)val);
+		break;
+	case HMC630X_IIO_ATTR_BB_ATTN1:
+	case HMC630X_IIO_ATTR_BB_ATTN2:
+		enum hmc6301_bb_attn attn1, attn2;
+		ret = hmc6301_get_bb_attn(d, &attn1, &attn2);
+		if (ret)
+			break;
+
+		if (priv == HMC630X_IIO_ATTR_BB_ATTN1)
+			ret = hmc6301_set_bb_attn(d, (enum hmc6301_bb_attn)val, attn2);
+		else
+			ret = hmc6301_set_bb_attn(d, attn1, (enum hmc6301_bb_attn)val);
+		break;
+	case HMC630X_IIO_ATTR_BB_ATTNI_FINE:
+	case HMC630X_IIO_ATTR_BB_ATTNQ_FINE:
+		enum hmc6301_bb_attn_fine attni, attnq;
+		ret = hmc6301_get_bb_attn_fine(d, &attni, &attnq);
+		if (ret)
+			break;
+
+		if (priv == HMC630X_IIO_ATTR_BB_ATTNI_FINE)
+			ret = hmc6301_set_bb_attn_fine(d, (enum hmc6301_bb_attn_fine)val, attnq);
+		else
+			ret = hmc6301_set_bb_attn_fine(d, attni, (enum hmc6301_bb_attn_fine)val);
+		break;
+	default:
+		return -EINVAL;
+	};
+
+	return ret;
+}
+
+static int hmc630x_iio_read_vco_available(void *device, char *buf,
+		uint32_t len, const struct iio_ch_info *channel,
+		intptr_t priv)
+{
+	int ret = 0;
+	struct hmc630x_iio_dev *iiodev = (struct hmc630x_iio_dev *)device;
+	struct hmc630x_dev *d = (struct hmc630x_dev *)iiodev->dev;
+	const uint64_t *freqs;
+	uint8_t n, nfreqs;
+	char fstr[12];
+
+	if (priv != HMC630X_IIO_ATTR_VCO_AVAILABLE)
+		return -EINVAL;
+
+	ret = hmc630x_get_avail_vco(d, &freqs, &nfreqs);
+	if (ret)
+		return ret;
+
+	buf[0] = '\0';
+	for (n = 0; n < nfreqs; n++) {
+		sprintf(fstr, "%lu ", freqs[n]/1000); // divide because too large for int
+		strcat(buf, fstr);
```

Comment:
You can write to `buf` directly: 

```
size_t len = 0;
len = sprintf(buf + len, "%llu ", freqs[n]/1000);
```
Also, the `%lu` modifier will only print 32 bits.

Reply:
will do in v2